viewing_party/party.py

```diff
@@ -1,23 +1,194 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
+def create_movie(movie_title, genre, rating):
+
+    movie= {}
+
+    if movie_title and genre and rating:
+    
+        movie["title"] = movie_title
+        movie["genre"] = genre
+        movie["rating"] = rating          
+        return movie
+
+    return None
+
+def add_to_watched(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watched": user_data["watched"][:]}
+
+    updated_user_data["watched"].append(movie)
+    return updated_user_data
+
 
-def create_movie(title, genre, rating):
-    pass
+def add_to_watchlist(user_data, movie):
+
+    # Create a copy of user data to avoid modifying original
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+        } 
+    updated_user_data["watchlist"].append(movie)
+    return updated_user_data
+
+
+def watch_movie(user_data, title):
+
+    # Create a copy of user data to avoid modifying original 
+    updated_user_data = {
+            "watchlist": user_data["watchlist"][:],
+            "watched": user_data["watched"][:]
+        } 
+    
+    # Check if movie with the title is in watchlist 
+    # and add it to the watched list
+    for movie in updated_user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_user_data["watchlist"].remove(movie)
+            updated_user_data["watched"].append(movie)         
+    return updated_user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    # User data is a dict which contains "watched" key
+    count = 0.0
+    sum = 0.0
+
+    if len(user_data["watched"]) == 0:
+        return sum
+    
+    # Count and sum rating values
+    for movie in user_data["watched"]:
+        if movie["rating"]:
+            count += 1
+            sum += movie["rating"]
+        else:
+            sum = 0
+
+    # Return average 
+    return sum/count
+
+
+def get_most_watched_genre(user_data):
+
+    max_value = 0
+    genre_frequencies = {}
+
+    if len(user_data["watched"]) == 0:
+        return None
+    
+    # Create key,value pairs {"genre" : genre_frequencies} in dict.
+    for movie in  user_data["watched"]:
+        if movie["genre"] in genre_frequencies:
+            genre_frequencies[movie["genre"]] += 1
+        else:
+            genre_frequencies[movie["genre"]] = 1

```

Comment:
How could we use the `.get` method so that we could condense this to a single line? 👀